dist/vendor.sh

```diff
@@ -1,4 +1,4 @@
-#!/usr/bin/env sh
+#!/usr/bin/env bash
```

Comment:
I am fine with `bash` but wondering what made you to switch?

Reply:
I usually use neovim for editing shell scripts but now I happened to open it in vscode with ShellCheck integration enabled. So I just followed the warnings given by it.

Reply:
`set -o pipefail` is a Bashism (it's not [in POSIX][1]).  That said, I see no pipes in this script today, so we could probably go with:

```sh
#!/usr/bin/env sh

set -ue
```

and see what vscode had to say about it ;).

[1]: https://pubs.opengroup.org/onlinepubs/9699919799/utilities/V3_chap02.html#set

Reply:
no need to introduce bash dependency

Reply:
> no need to introduce bash dependency

You can argue with shellcheck about this:

```console
$ shellcheck dist/vendor.sh

In dist/vendor.sh line 6:
FILES_TO_COMMIT=(
                ^-- SC2039: In POSIX sh, arrays are undefined.


In dist/vendor.sh line 15:
  mv --backup=existing ${CARGO_CONFIG}{,.prev}
                                      ^------^ SC2039: In POSIX sh, brace expansion is undefined.


In dist/vendor.sh line 32:
  git add --force "${FILES_TO_COMMIT[@]}"
                   ^-------------------^ SC2039: In POSIX sh, array references are undefined.


In dist/vendor.sh line 33:
  git diff --quiet --staged -- "${FILES_TO_COMMIT[@]}" || {
                                ^-------------------^ SC2039: In POSIX sh, array references are undefined.


In dist/vendor.sh line 35:
    git commit -m "Update vendored dependencies and rh-manifest.txt" -- "${FILES_TO_COMMIT[@]}"
                                                                         ^-------------------^ SC2039: In POSIX sh, array references are undefined.

For more information:
  https://www.shellcheck.net/wiki/SC2039 -- In POSIX sh, array references are...
```

These warnings all go away if I change the shebang to refernece `bash`.

Reply:
These are issues to fix, not a requirement to have bash

Reply:
They are features of bash and other shells and were only displayed by shellcheck because of the `sh` shebang. So the fix is simply to switch to bash, since I think depending on bash is a very reasonable requirement. How do you justify the restriction your suggesting?

Reply:
I can not think of any downside of switching to bash. Also I am fine with bash as it will reduce the effort we need to put to fix issues if we want to stick to `sh`

Reply:
Can we just drop `FILES_TO_COMMIT`?  It is an attempt to allow the vendor script to run in the face of an otherwise dirty working directory without committing the dirt, but it's not clear to me that folks would want to do that (and `git stash ...` provides a fairly convenient workaround if they do have dirt they don't want committed).  If we want to keep it and stick with POSIX, we could use `set` and `${@}` to maintain this array.